resolwe/flow/serializers/relation.py

```diff
@@ -47,6 +47,13 @@ class Meta:
             'unit',
         )
 
+    def validate_partitions(self, partitions):
+        """Raise validation error if list of partitions is empty."""
+        if not partitions:
+            raise serializers.ValidationError("List of partitions must not be empty.")
```

Comment:
Are we sure we want this?

Reply:
Yes, we automatically delete Relation when the last entity is removed, so we don't want to allow to create such "invalid" state.

Reply:
Hm, thinking about this more, @hadalin may have an issue with this. The user workflow is such: user clicks a button to create a relation. Then the user selects some sample to create a group on this collection, then another group. User may remove a group from collection. Or change groups if she likes...

So, when the user creates a collection, Primoz will not be able to create it in the backend yet (until the user creates at least one group). So if the user navigates away and back, she will lose her work—the "empty" collection will disappear.

Reply:
Maybe we don't want to delete empty collections automatically after all.

Reply:
I worry this really complicates the implementation in the web frontend.

Reply:
The workflow is as such: when a user creates a new relation, it is created together with groups (partitions). A user can remove all relations on a collection or a particular relation (a table column), but not groups. So if the only concern is the list of partitions "must not be empty" then I don't think we have a problem.

Reply:
> So if the only concern is that the list of partitions "must not be empty", then I don't think we have a problem.

That was my only concern, thanks!